### ISSUE TITLE
How to handle errors occured during the GraphQL resolver execution?

### ISSUE THREAD

**Issue by mistyharsh** [fsprojects/FSharp.Data.GraphQL#299]
### Description

Error handling documentation is missing/not very clear. I found `AddError` method on `Context` object but it does not seem to produce any result.

### Repro steps

As the title of the issue suggest, how to properly handle errors? The error is described here in [detail on stackoverflow](https://stackoverflow.com/q/63665037/5723098).

### Expected behavior

Ideally I would like to generate errors as:

```
{
    "documentId": 1234,
    "data": {
        "login": null
    },
    
    // The errors array is not available
    "errors": [
        { message: "error description", code: 1001, path: "" } 
    ]
}
```

### Actual behavior

  - `AddError` method doesn't seem to provide any functionality.
  - Errors raised via exception are not customizable in final response.

**Comment by mickhansen** [comment]
Is there any way to produce an error response without having to `raise`?

**Comment by mickhansen** [comment]
`ctx.AddError` plus middleware that adds the errors to the response works decently, but you lose `path`.
It'd be great to have `path` exposed in `ResolveFieldContext`

**Comment by xperiandri** [comment]
@mistyharsh now you can return `IGQLError` interface optionally implementing `IGQLErrorExtensions` interface to attach extensions according to specification.
Aslo, you get very detailed errors out of the box now. See #440

**Comment by xperiandri** [comment]
For exception handling see StarWarsAPI project

**Comment by xperiandri** [comment]
Fixed in 2.0.0
